scipy/signal/_filter_design.py

```diff
@@ -1778,30 +1801,37 @@ def normalize(b, a):
     Badly conditioned filter coefficients (numerator): the results may be meaningless
 
     """
-    num, den = b, a
+    try:
+        xp = array_namespace(b, a)
+    except TypeError:
+        # object arrays, test_ltisys.py::TestSS2TF::test_simo_round_trip
+        xp = np_compat
 
-    den = np.asarray(den)
-    den = np.atleast_1d(den)
-    num = np.atleast_2d(_align_nums(num))
+    den = xp.asarray(a)
+    den = xpx.atleast_nd(den, ndim=1, xp=xp)
+
+    num = xp.asarray(b)
+    num = xpx.atleast_nd(_align_nums(num, xp), ndim=2, xp=xp)
 
     if den.ndim != 1:
         raise ValueError("Denominator polynomial must be rank-1 array.")
     if num.ndim > 2:
         raise ValueError("Numerator polynomial must be rank-1 or"
                          " rank-2 array.")
-    if np.all(den == 0):
+    if xp.all(den == 0):
         raise ValueError("Denominator must have at least on nonzero element.")
 
     # Trim leading zeros in denominator, leave at least one.
-    den = np.trim_zeros(den, 'f')
+    den = _trim_zeros(den, 'f')
 
     # Normalize transfer function
     num, den = num / den[0], den / den[0]
 
     # Count numerator columns that are all zero
     leading_zeros = 0
-    for col in num.T:
-        if np.allclose(col, 0, atol=1e-14):
+    for j in range(num.shape[-1]):
+        col = num[:, j]
+        if xp.all(xp.abs(col) <= 1e-14):
```

Comment:
Is this equivalent since `rtol` has a default value?

Reply:
In general it isn't indeed. Here specifically, the second argument of `np.allclose`, `b`, is zero. And `np.allclose` is defined for `rtol` relative to `b`:

```
absolute(a - b) <= (atol + rtol * absolute(b))
```
https://numpy.org/doc/stable/reference/generated/numpy.allclose.html